lib/avo/fields/belongs_to_field.rb

```diff
@@ -229,22 +238,22 @@ def database_id
         id
       end
 
-      def target_resource
+      def target_resource(record: @record, polymorphic_model_class: value&.class)
         @target_resource ||= if use_resource.present?
```

Comment:
This is not right.
We can't memoize a value in a method if the method takes arguments.

If one time the `record` is a `User`, it will get memoized as that user. Then, the second time when the `record` is a `team`, it will return the first user, resulting in a bug.

Did I read that correctly?